src/common.rkt

```diff
@@ -104,6 +101,81 @@
   (check-true (set-disjoint? '() '()))
   (check-false (set-disjoint? '(a b c) '(a))))
 
+(define (list-suffix? l r)
```

Comment:
I think we should not have the code that uses these and then also we won't need these.